events/upstream_push.go

```diff
@@ -0,0 +1,138 @@
+			if err := tx.Where("id IN ?", itemIDs).Find(&upstreamMsg.ConfigItems).Error; err != nil {
+				return fmt.Errorf("error fetching config_items: %w", err)
+			}
+
+		case "check_statuses":
+			if err := tx.Where(`(check_id, "time") IN ?`, itemIDs).Find(&upstreamMsg.CheckStatuses).Error; err != nil {
+				return fmt.Errorf("error fetching check_statuses: %w", err)
+			}
+
+		case "config_component_relationships":
+			if err := tx.Where("(component_id, config_id) IN ?", itemIDs).Find(&upstreamMsg.ConfigItems).Error; err != nil {
+				return fmt.Errorf("error fetching config_component_relationships: %w", err)
+			}
+
+		case "component_relationships":
+			if err := tx.Where("(component_id, relationship_id, selector_id) IN ?", itemIDs).Find(&upstreamMsg.ComponentRelationships).Error; err != nil {
+				return fmt.Errorf("error fetching component_relationships: %w", err)
+			}
+
+		case "config_relationships":
+			if err := tx.Where("(related_id, config_id, selector_id) IN ?", itemIDs).Find(&upstreamMsg.ConfigRelationships).Error; err != nil {
+				return fmt.Errorf("error fetching config_relationships: %w", err)
+			}
+		}
+	}
+
+	if err := t.push(ctx, upstreamMsg); err != nil {
+		return fmt.Errorf("failed to push to upstream: %w", err)
+	}
+
+	return nil
+}
+
+func (t *pushToUpstreamEventHandler) push(ctx context.Context, msg *api.PushData) error {
+	payloadBuf := new(bytes.Buffer)
+	if err := json.NewEncoder(payloadBuf).Encode(msg); err != nil {
+		return fmt.Errorf("error encoding msg: %w", err)
+	}
+
+	req, err := http.NewRequest(http.MethodPost, t.conf.URL, payloadBuf)
+	if err != nil {
+		return fmt.Errorf("http.NewRequest: %w", err)
+	}
+
+	req.SetBasicAuth(t.conf.Username, t.conf.Password)
+	resp, err := t.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("error making request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusCreated && resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("upstream server returned error status: %s", resp.Status)
+	}
+
+	return nil
+}
+
+func groupChangelogsByTables(events []api.Event) map[string][]any {
+	var output = make(map[string][]any)
+	for _, cl := range events {
+		tableName := cl.Properties["table"]
+		switch tableName {
+		case "component_relationships":
+			output[tableName] = append(output[tableName], []string{cl.Properties["component_id"], cl.Properties["relationship_id"], cl.Properties["selector_id"]})
+		case "config_component_relationships":
+			output[tableName] = append(output[tableName], []string{cl.Properties["component_id"], cl.Properties["config_id"]})
+		case "config_relationships":
+			output[tableName] = append(output[tableName], []string{cl.Properties["related_id"], cl.Properties["config_id"], cl.Properties["selector_id"]})
+		case "check_statuses":
+			output[tableName] = append(output[tableName], []string{cl.Properties["check_id"], cl.Properties["time"]})
+		default:
+			output[tableName] = append(output[tableName], cl.Properties["id"])
```

Comment:
Set a variable for switch case and run `output[tablename] = append(..., var` only once

Reply:
Can you please elaborate on this?

Reply:
```golang
		switch tableName {
		case "component_relationships":
			logs = []string{cl.Properties["component_id"], cl.Properties["relationship_id"], cl.Properties["selector_id"]})
		case "config_component_relationships":
			logs = []string{cl.Properties["component_id"], cl.Properties["config_id"]})
		case "config_relationships":
        ...
		default:
			logs = cl.Properties["id"])
    output[tablename] = append(..., logs)
```